### PR TITLE
[ADF-3308] Process list filter by fields

### PR DESCRIPTION
**Pull request by bbcodrin**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [X] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
We cannot filter the process list.
https://issues.alfresco.com/jira/browse/ADF-3308

**What is the new behaviour?**
We add a page that shows the process list with an input text for each of the some properties.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell/resources/i18n/en.json [hunk 1]
```diff
@@ -208,5 +209,9 @@
   "TASK_LIST_DEMO": {
     "APP_ID_REQUIRED_ERROR": "Insert App Id",
     "APP_ID_TYPE_ERROR": "App Id must be a number"
+  },
+  "PROCESS_LIST": {
```
Comment (mauriziovitale) [inline demo-shell/resources/i18n/en.json]: can you call it 
PROCESS_LIST_DEMO

File: demo-shell/src/app/app.module.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ import { DemoPermissionComponent } from './components/permissions/demo-permissio
 import { PreviewService } from './services/preview.service';
 import { BreadcrumbDemoComponent } from './components/breadcrumb-demo/breadcrumb-demo.component';
 import { TaskListDemoComponent } from './components/task-list-demo/task-list-demo.component';
+import { ProcessListComponent } from './components/process-list/process-list.component';
```
Comment (mauriziovitale) [inline demo-shell/src/app/app.module.ts]: ProcessListDemoComponent
process-list-demo.component

File: docs/process-services/process-list.component.md [hunk 3]
```diff
@@ -49,6 +49,7 @@ Renders a list containing all the process instances matched by the parameters sp
 | page | `number` | 0 | The page number of the processes to fetch. |
 | presetColumn | `string` |  | Name of a custom schema to fetch from `app.config.json`. |
 | processDefinitionKey | `string` |  | The processDefinitionKey of the process. |
+| processDefinitionId | `string` |  | The Definition Id of the process. |
```
Comment (mauriziovitale) [inline docs/process-services/process-list.component.md]: the component doesn't expose this input 
```javascript
@Input()
    appId: number;

    /** The processDefinitionKey of the process. */
    @Input()
    processDefinitionKey: string;

    /** Defines the state of the processes. Possible values are `running`, `completed` and `all` */
    @Input()
    state: string;

    /** Defines the sort ordering of the list. Possible values are `created-desc`, `created-asc`,
     * `ended-desc`, `ended-asc`.
     */
    @Input()
    sort: string;

    /** The name of the list. */
    @Input()
    name: string;

    /** The page number of the processes to fetch. */
    @Input()
    page: number = 0;

    /** The number of processes to fetch in each page. */
    @Input()
    size: number = PaginationComponent.DEFAULT_PAGINATION.maxItems;
```
why we are adding this to the doc?

File: lib/process-services/process-list/components/process-list.component.ts [hunk 4]
```diff
@@ -154,7 +154,7 @@ export class ProcessInstanceListComponent implements OnChanges, AfterContentInit
         this.loadLayoutPresets();
         this.setupSchema();
 
-        if (this.appId) {
+        if (this.appId != null) {
```
[the remark this answers is not in the document]
Comment (bbcodrin) [reply lib/process-services/process-list/components/process-list.component.ts]: By default 'this.appId' is a number with 0 value. So we need to change the condition so we can call this.reload() even if 0 is the value.

**Comment by mauriziovitale** [comment]
e2e test failing are related to a different component

**Pull request merged**